src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,172 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import core.basesyntax.service.InvalidDataException;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceTest {
+    private RegistrationServiceImpl registrationService;
+
+    @BeforeEach
+    void setUp() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @AfterEach
+    void cleanUp() {
+        Storage.PEOPLE.clear();
+    }
+
+    @Test
+    public void register_validUser_successfulRegistration() {
+        User user = new User();
+        user.setLogin("username");
+        user.setPassword("password123");
+        user.setAge(20);
+
+        User registeredUser = registrationService.register(user);
+        assertNotNull(registeredUser);
+        assertEquals(user.getLogin(), registeredUser.getLogin());
+    }
+
+    @Test
+    public void register_existingUser_throwException() {
+        User user1 = new User();
+        user1.setLogin("username");
+        user1.setPassword("password123");
+        user1.setAge(20);
+
+        registrationService.register(user1);
```

Comment:
better not to use register()  for preparation testing data when you test this method. You can add users to storage directly.